.github/workflows/release_java.yml

```diff
@@ -60,6 +62,15 @@ jobs:
         with:
           version: "23.4"
           repo-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Setup linux-aarch_64
+        if: "contains(matrix.classifier, 'linux-aarch_64')"
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y gcc-aarch64-linux-gnu g++-aarch64-linux-gnu
+          # Setup for cargo
+          echo "CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER=aarch64-linux-gnu-gcc" >> $GITHUB_ENV
```

Comment:
Amazon Linux 2 has glibc 2.26 while `ubuntu-latest` (currently Ubuntu 22.04) has glibc 2.35, so my gut feeling is that this cross compiled version won't load on AL2.

Reply:
> Amazon Linux 2 has glibc 2.26 while `ubuntu-latest` (currently Ubuntu 22.04) has glibc 2.35, so my gut feeling is that this cross compiled version won't load on AL2.

But the x86_64 version works, I thought they are using the same glibc version?

Reply:
Hi, @amunra, would you like to test the jar we built? You can find the linux-aarch64 jars here: https://github.com/apache/incubator-opendal/actions/runs/6976190245

Reply:
I'm not sure, never tried AL2. Did the bug reporter uses the rleased version or did they compiled their own version?

Reply:
From this: https://github.com/apache/incubator-opendal/issues/3629#issue-2002065970 I assume they compiled it on a AL2 compatible system?

Reply:
The `.so` definitely requires newer glibc version than 2.26.

```bash
$ readelf -s libopendal_java.so | grep GLIBC | grep -v GLIBCXX | grep -v 2.17
     8: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND stat@GLIBC_2.33 (4)
    14: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND exp@GLIBC_2.29 (7)
    15: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND f[...]@GLIBC_2.28 (8)
    17: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND log@GLIBC_2.29 (7)
    24: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.34 (10)
    32: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.34 (10)
    38: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.30 (13)
    49: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.34 (10)
    63: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.34 (10)
    72: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.34 (10)
    74: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND fstat@GLIBC_2.33 (4)
    93: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.34 (10)
    97: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.34 (10)
   118: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.34 (10)
   119: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.34 (10)
   127: 0000000000000000     0 FUNC    WEAK   DEFAULT  UND statx@GLIBC_2.28 (8)
   151: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.34 (10)
   154: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND l[...]@GLIBC_2.33 (4)
   167: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.34 (10)
   178: 0000000000000000     0 FUNC    WEAK   DEFAULT  UND [...]@GLIBC_2.25 (22)
   181: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.34 (10)
   185: 0000000000000000     0 FUNC    WEAK   DEFAULT  UND [...]@GLIBC_2.27 (23)
   196: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND stat64@GLIBC_2.33 (4)
   202: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.34 (10)
   235: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.34 (10)
   238: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.34 (10)
   245: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.34 (10)
   256: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.34 (10)
   274: 0000000000000000     0 FUNC    WEAK   DEFAULT  UND [...]@GLIBC_2.18 (29)
   286: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.34 (10)
   288: 0000000000000000     0 OBJECT  GLOBAL DEFAULT  UND [...]@GLIBC_2.32 (30)
   306: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.34 (10)
   315: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND f[...]@GLIBC_2.33 (4)
   326: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.34 (10)
   330: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND dlsym@GLIBC_2.34 (10)
   335: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.34 (10)
   341: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.34 (10)
   346: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND pow@GLIBC_2.29 (7)
   366: 0000000000000000     0 FUNC    WEAK   DEFAULT  UND [...]@GLIBC_2.29 (34)
   371: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND f[...]@GLIBC_2.33 (4)
   388: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND log2@GLIBC_2.29 (7)
   391: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.34 (10)
   416: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.34 (10)
   425: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND [...]@GLIBC_2.32 (30)
340121: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND stat@GLIBC_2.33
340127: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND exp@GLIBC_2.29
340128: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND fcntl64@GLIBC_2.28
340130: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND log@GLIBC_2.29
340154: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND gettid@GLIBC_2.30
340191: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND dlopen@GLIBC_2.34
340193: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND fstat@GLIBC_2.33
340213: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND dlclose@GLIBC_2.34
340250: 0000000000000000     0 FUNC    WEAK   DEFAULT  UND statx@GLIBC_2.28
340279: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND lstat64@GLIBC_2.33
340309: 0000000000000000     0 FUNC    WEAK   DEFAULT  UND getrandom@GLIBC_2.25
340329: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND stat64@GLIBC_2.33
340372: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND dlerror@GLIBC_2.34
340455: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND fstatat64@GLIBC_2.33
340470: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND dlsym@GLIBC_2.34
340487: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND pow@GLIBC_2.29
340512: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND fstat64@GLIBC_2.33
340530: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND log2@GLIBC_2.29
```

Reply:
Any idea to fix this?

Reply:
Use an old toolchain...

See also https://peps.python.org/pep-0513/#key-causes-of-inter-linux-binary-incompatibility

Reply:
Tracked at https://github.com/apache/incubator-opendal/issues/3663